CMakeLists.txt

```diff
@@ -221,11 +221,13 @@ if (ENABLE_EXPENSIVE_CHECKS)
     add_definitions("-DAD_ENABLE_EXPENSIVE_CHECKS")
 endif()
 
-if (DISABLE_QUERY_CANCELLATION_WATCH_DOG)
-    message(STATUS "Disable periodic checks that the cancellation flag is checked frequently enough while the result of
-    a query is being computed with a small runtime overhead")
-    add_definitions("-DQLEVER_DISABLE_QUERY_CANCELLATION_WATCH_DOG")
+if (NOT QUERY_CANCELLATION_MODE)
+    set(QUERY_CANCELLATION_MODE ENABLED CACHE STRING "The loglevel")
```

Comment:
1. Whatever the `STRING` does here, it is out of sync.
2. Is it possible to specify the `ENABLED` directly below in the `set_property` command (I'm not a CMake expert, so this is a mutual research endeavor.